src/hybrid/vector_query_utils.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+#ifndef VECTOR_QUERY_UTILS_H
+#define VECTOR_QUERY_UTILS_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include "../vector_index.h"
+#include "../query_node.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct ParsedVectorQuery {
```

Comment:
Don't we have such a struct already, that can be used, even if partially full?
If it's very different fine, but I suspect it's quite similar..